helper/keystore/keystore.go

```diff
@@ -11,30 +11,44 @@ type createFn func() ([]byte, error)
 
 type readFn func(b []byte) (interface{}, error)
 
+// CreateIfNotExists generates a private key at the specified path,
+// or reads it if a key file is present
 func CreateIfNotExists(path string, create createFn, read readFn) (interface{}, error) {
 	_, err := os.Stat(path)
 	if err != nil && !os.IsNotExist(err) {
 		return nil, fmt.Errorf("failed to stat (%s): %v", path, err)
 	}
+
+	var keyBuff []byte
 	if os.IsNotExist(err) {
-		// create the file
-		buf, err := create()
+		// Key doesn't exist yet
+
+		// Create a key and write it to disk as base64
+		keyBuff, err = create()
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("unable to generate private key, %v", err)
 		}
-		if err := ioutil.WriteFile(path, []byte(hex.EncodeToString(buf)), 0600); err != nil {
-			return nil, err
+
+		// Encode it to a readable format (Base64)
+		keyBuff = []byte(hex.EncodeToString(keyBuff))
+		if err = ioutil.WriteFile(path, keyBuff, 0600); err != nil {
+			return nil, fmt.Errorf("unable to write private key to disk (%s), %v", path, err)
 		}
-	}
+	} else {
+		// Key exists
 
-	// read the file
-	raw, err := ioutil.ReadFile(path)
-	if err != nil {
-		return nil, err
+		// Read the Base64 encoded key from the disk.
+		// Base64 decoding is done in the read method
+		keyBuff, err = ioutil.ReadFile(path)
+		if err != nil {
+			return nil, fmt.Errorf("unable to read private key from disk (%s), %v", path, err)
+		}
 	}
```

Comment:
You could get rid of if/else and just do an early return of the key read from disk if it exists. In my opinion it's more readable that way than having to go back and to see what the initial `if` was for in whose `else` branch we are in.

Reply:
You're right. It's probably unnecessarily complicated. 

I've simplified it a bit, and moved out the common read part out to `crypto.go` 👍